src/controller/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  if (!userId && !categories && (!from || !to)) {
+    res.send(expensesService.getAllExpenses());
+  }
+
+  const normalizedCategories = expensesService.normalizeCategory(categories);
+
+  let filteredExpenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(userId);
+  }
+
+  if (normalizedCategories) {
+    filteredExpenses =
+      expensesService.filterExpensesByCategory(normalizedCategories);
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(fromDate, toDate);
+  }
+
+  res.send(filteredExpenses);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+
+  const expense = expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount || !category || !note) {
+    return res.sendStatus(400);
+  }
+
+  const user = userService.getById(userId);
+
+  if (!user) {
+    res.sendStatus(400);
+
+    return;
+  }
```

Comment:
This condition checks if the user exists. If not, it returns a 400 status. However, it would be more accurate to return a 404 status, as it's the client's error of providing a non-existing `userId`.